cluster/gce/upgrade.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# !!!EXPERIMENTAL !!! Upgrade script for GCE. Expect this to get
+# rewritten in Go in relatively short order, but it allows us to start
+# testing the concepts.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+readonly CI_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-(.*)$"
+
+if [[ "${KUBERNETES_PROVIDER:-gce}" != "gce" ]]; then
+  echo "!!! ${1} only works on GCE" >&2
+  exit 1
+fi
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/kube-env.sh"
+source "${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
+
+function usage() {
+  echo "!!! EXPERIMENTAL !!!"
+  echo ""
+  echo "${0} [-M|-n] <release or continuous integration version>"
+  echo "  Upgrades all nodes by default"
```

Comment:
I still dislike that we use the word "nodes". It throws me all the time, because nodes, to me, is a generic word for entities in a cluster, so I meant nodes (machine in a cluster), not nodes (minion) here, but that clearly doesn't work. :) Fixed to use our rather confusing definition.